chevah/compat/posix_filesystem.py

```diff
@@ -384,6 +385,32 @@ def openFileForWriting(self, segments, utf8=False):
                     DEFAULT_FILE_MODE)
                 return os.fdopen(fd, 'wb')
 
+    def openFileForUpdating(self, segments, utf8=False):
+        """
+        See `ILocalFilesystem`.
+        """
+        path = self.getRealPathFromSegments(segments)
+        path_encoded = self.getEncodedPath(path)
+
+        self._requireFile(segments)
+        with self._IOToOSError(path), self._impersonateUser():
+            if utf8:
+                # This will also allow reading, but I hope we will
+                # remove this API soon, and we patch the high level API.
+                file = codecs.open(path_encoded, 'r+', 'utf-8')
+
+                def deny_read(size=None):
+                    raise IOError('File not open for reading')
+
+                file.read = deny_read
+                return file
+            else:
+                fd = os.open(
+                    path_encoded,
+                    self.OPEN_WRITE_ONLY,
+                    DEFAULT_FILE_MODE)
+                return os.fdopen(fd, 'wb')
```

Comment:
> Please note that for the os.fdopen(fd, 'wb') the FD is already created

I noticed that as well but the descriptor is obtained via `open(path, OPEN_WRITE_ONLY)` which according to the documentation found will truncate.

My opinion is that the documentation is not ok; it's from a post on Stack Exchange anyway. When testing locally I used `os.open` with `O_WRONLY`.